### PR TITLE
fix(middleware): add HEAD method support

### PR DESCRIPTION
**Pull request by EslamHiko**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
yes

**Summary**
fixes: https://github.com/webpack/webpack-dev-server/issues/1833

**Does this PR introduce a breaking change?**
no

**Other information**
when I run tests I get lint errors : 
`Expected linebreaks to be 'LF' but found 'CRLF'` and  `Delete '␍'`
and when I disable lint and run tests only requests' tests fail in expecting `Content-Length` and `Content-Range`
I'm using windows and I tried this on Ubuntu and it's the same. so is it normal?

### CODE REVIEW

File: lib/middleware.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ module.exports = function wrapper(context) {
       });
     }
 
-    const acceptedMethods = context.options.methods || ['GET'];
+    const acceptedMethods = context.options.methods || ['GET', 'HEAD'];
```
Comment (evilebottnawi) [inline lib/middleware.js]: What about support all method by defaults?

**Comment by evilebottnawi** [comment]
Let's wait CI green and release as new version

Honestly, I don’t know why before that we only allowed `get` :confused:

**Comment by EslamHiko** [comment]
@evilebottnawi I think because the most used method of requests is `get` so it was set to be the default. the problem was allowing `get` only.

File: README.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ _Note: The `publicPath` property is required, whereas all other options are opti
 ### methods
 
 Type: `Array`  
-Default: `[ 'GET' ]`
+Default: `undefined`
```
Comment (hiroppy) [inline README.md]: I think this change seems to be breaking change. Is this wrong?

**Comment by hiroppy** [comment]
> Honestly, I don’t know why before that we only allowed get

Yeah, me too. So I think this change is good.

File: README.md (hunk 2 above)
Comment (EslamHiko) [reply README.md]: @hiroppy I don't think so here is the check : 
```
const acceptedMethods = context.options.methods;

    if (acceptedMethods && acceptedMethods.indexOf(req.method) === -1) {
      return goNext();
    }
```
and in tests I changed the `post` request test that supposed to `return 404` in the previous settings to be like `get` now it passes the test without issues. 

I changed the documentation a little in the last commit, so now it makes more sense.

Comment (evilebottnawi) [reply README.md]: Yes, it is not breaking change, it is feature

**Comment by evilebottnawi** [comment]
After investigate in other middleware code (one of popular exmaples https://github.com/expressjs/serve-static/blob/master/index.js#L73) i find what i was wrong, we should handle only `GET` and `HEAD` by default

**Comment by evilebottnawi** [comment]
I send a PR, sorry for this

**Comment by evilebottnawi** [comment]
Close in favor https://github.com/webpack/webpack-dev-middleware/pull/398, sorry again for misleading

**Pull request closed without merging**